torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -676,6 +676,12 @@ void hashValue(Stack* stack) {
   push(stack, int64_t(hash));
 }
 
+void hashGenericIValue(Stack* stack) {
+  auto value = pop(stack);
+  auto hash = (int64_t)IValue::HashAliasedIValue()(value);
```

Comment:
This hash is only suppose to work for aliasing ivalues: https://github.com/pytorch/pytorch/blob/b44a10c1791f3f1d27171e6d012ca5f661e58aa9/aten/src/ATen/core/ivalue.h#L679
in python immutable scalars are hashed by value:
`hash((1, 2)) == hash((1, 2))` which this doesnt do

Reply:
Ah, okay. So I guess I'll define a canonical `IValue::hash`